### ISSUE TITLE
libvmui.textinput() doesn't support moving curser with mouse.

### ISSUE THREAD

**Issue by ThomasTheSpaceFox** [SBTCVM/SBTCVM-Mark-2#1]
The recently added general-use text input function does feature a movable cursor, but its only movable via the keyboard. the question is how to add mouse support without massively over-complicating it.

it would help the general usability of the calculator a bit, as well as any utilities that end up using libvmui.textinput() as well.

**Comment by ThomasTheSpaceFox** [comment]
@SBTCVM/core-devs any ideas?

**Comment by wyatthuckaby** [comment]
ill take a look, is the issue in the current branch?

**Comment by ThomasTheSpaceFox** [comment]
@wyatthuckaby yup. perhaps the size() method of pygame font objects might be of use? to determine the size of the characters. see pygame docs: (http://www.pygame.org/docs/ref/font.html#pygame.font.Font.size)  the new calculator (calc.py) is the first utility to use it. (and its handy character mask feature) also, MK2-TOOLS.py has a test for it via the  "textinput" argument.

**Comment by ThomasTheSpaceFox** [comment]
i should also point out that the cursor is an actual underscore/space that is inserted into the a copy of the current string that is used for rendering. this will affect the position of characters a bit.

**Comment by ThomasTheSpaceFox** [comment]
Fixing this in the desktop shell builtin should not be much more work, given that the code actually functions the same way as libvmui.textinput() just with slightly modified event parsing, and a different rendering scenario...
